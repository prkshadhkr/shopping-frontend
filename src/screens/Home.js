import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../user/UserContext';
import './Home.css';

const Home =() =>{
	const { user } = useContext( UserContext );
	return (
		<div className="Home d-flex justify-content-center">
			<div >
				<h1 className="d-flex justify-content-center">Shopping</h1>
				<p className="d-flex justify-content-center font-weight-light">
					Onestop shop: Checkout our great products!</p>
				{!user ?(
					<div className="Login d-flex justify-content-center">
						<Link 
						  className="btn btn-outline-primary"
						  to="/login" >
						    Login
					  </Link>
					</div>):(
				<h5 className="d-flex justify-content-center">
					Welcome back {user.first_name}!
				</h5>
				)
			}
			</div>
		</div>
	)
}

export default Home;