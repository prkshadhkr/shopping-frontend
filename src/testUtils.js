import React from  'react';
import UserContext from './user/UserContext';

const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { UserProvider };