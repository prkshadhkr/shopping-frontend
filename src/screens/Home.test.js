import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../testUtils';

it ('renders without crashing', function() {
  render(
  <MemoryRouter >
    <UserProvider>
      <Home />
    </UserProvider>
  </MemoryRouter>);
});

it ('matches snapshot', function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})

it('should match snapshot if user logout', function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider user={null}>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})