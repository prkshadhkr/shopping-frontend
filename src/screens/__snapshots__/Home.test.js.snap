// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home d-flex justify-content-center"
  >
    <div>
      <h1
        class="d-flex justify-content-center"
      >
        Shopping
      </h1>
      <p
        class="d-flex justify-content-center font-weight-light"
      >
        Onestop shop: Checkout our great products!
      </p>
      <h5
        class="d-flex justify-content-center"
      >
        Welcome back testing!
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot if user logout 1`] = `
<DocumentFragment>
  <div
    class="Home d-flex justify-content-center"
  >
    <div>
      <h1
        class="d-flex justify-content-center"
      >
        Shopping
      </h1>
      <p
        class="d-flex justify-content-center font-weight-light"
      >
        Onestop shop: Checkout our great products!
      </p>
      <div
        class="Login d-flex justify-content-center"
      >
        <a
          class="btn btn-outline-primary"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
